service/worker/esanalyzer/workflow.go

```diff
@@ -171,6 +216,123 @@ func (w *Workflow) emitWorkflowVersionMetrics(ctx context.Context) error {
 	return nil
 }
 
+func (w *Workflow) emitWorkflowVersionMetricsPinot(domainName string, logger *zap.Logger) error {
+	wfVersionPinotQuery, err := w.getWorkflowTypePinotQuery(domainName)
+	if err != nil {
+		logger.Error("Failed to get Pinot query to find workflow version Info",
+			zap.Error(err),
+			zap.String("DomainName", domainName),
+		)
+		return err
+	}
+	response, err := w.analyzer.pinotClient.SearchAggr(&pinot.SearchRequest{Query: wfVersionPinotQuery})
+	if err != nil {
+		logger.Error("Failed to query Pinot to find workflow type count Info",
+			zap.Error(err),
+			zap.String("VisibilityQuery", wfVersionPinotQuery),
+			zap.String("DomainName", domainName),
+		)
+		return err
+	}
+	foundAggregation := len(response) > 0
+
+	if !foundAggregation {
+		logger.Error("Pinot error: aggregation failed.",
+			zap.Error(err),
+			zap.String("Aggregation", fmt.Sprintf("%v", response)),
+			zap.String("DomainName", domainName),
+			zap.String("VisibilityQuery", wfVersionPinotQuery),
+		)
+		return err
+	}
+	var domainWorkflowVersionCount DomainWorkflowVersionCount
+	for _, row := range response {
```

Comment:
10x latency might be an issue for metrics emission. Could you parallelize it?

Reply:
If we do this in parallel with multiple threads, is there a risk when metrics are emitted, the workflow still doesn't have all the data?

Reply:
This metrics doesn't care about the latency, since we run it every 5 or 10 minutes. But we can eliminate the calls when we aggregate by both version and type

Reply:
Discussed this with Ender as well. We are going to keep this approach.